src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1389,38 +1398,34 @@ void Connection::receive_test_command_response(session_ident_type session_ident,
         return;
     }
 
-    if (auto ec = sess->receive_test_command_response(request_ident, body)) {
-        close_due_to_protocol_error(ec);
+    if (auto status = sess->receive_test_command_response(request_ident, body); !status.is_ok()) {
+        close_due_to_protocol_error(std::move(status));
     }
 }
 
 
-void Connection::handle_protocol_error(ClientProtocol::Error error)
+void Connection::handle_protocol_error(ClientProtocol::Error error, std::string msg)
```

Comment:
Thanks for adding this!
IMHO - all of our error handlers should include some sort of error message for context and these changes (particularly for using `Status`) are moving toward that.